include/revng/Model/CommonTypeMethods.h

```diff
@@ -0,0 +1,224 @@
+  bool isConst() const;
+
+  /// Skip all the "wrappers" around the definition.
+  ///
+  /// This is provided for some very specific cases, please take extra care when
+  /// using it. If you are hesitating whether you need this, please shop around
+  /// other functions provided here. If any of them sounds like it should do
+  /// the trick, prefer it instead.
+  ///
+  /// \returns a pointer to the definition, or a `nullptr` for primitive types
+  ///          that do not have one.
+  ///
+  /// \note in the cases when this returns `nullptr`, the type does not have
+  ///       a definition, as in, the inner-most node is a primitive.
+  model::TypeDefinition *skipToDefinition();
+  const model::TypeDefinition *skipToDefinition() const;
+
+  /// An even more specialized version of \ref skipToDefinition.
+  ///
+  /// It only differs in the sense that the found `DefinedType` is returned as
+  /// is instead of getting automatically unwrapped. Which bring inability
+  /// to call it on a `model::TypeDefinition`: it will abort.
+  model::DefinedType *skipToDefinedType();
+  const model::DefinedType *skipToDefinedType() const;
+
+public:
+  /// Returns the pointee type while unwrapping typedefs.
+  ///
+  /// \note This asserts if the type is _not_ a pointer, so only use it when
+  ///       it's guaranteed to be one.
+  ///
+  /// \note This also ignores the constness of the dropped pointer (and typedefs
+  ///       in the way)
+  model::Type &stripPointer();
+  const model::Type &stripPointer() const;
+
+public:
+  /// Ensures the type this called onto is not const.
+  ///
+  /// \note This can modify some typedefs within the model: use with care.
+  void stripConstness();
+
+private:
+  using PKind = model::PrimitiveKind::Values;
+
+public:
+  bool isPrimitive(model::PrimitiveKind::Values Kind) const;
+  bool isVoid() const { return isPrimitive(PKind::Void); }
+  bool isGeneric() const { return isPrimitive(PKind::Generic); }
+  bool isPointerOrNumber() const { return isPrimitive(PKind::PointerOrNumber); }
+  bool isNumber() const { return isPrimitive(PKind::Number); }
+  bool isUnsigned() const { return isPrimitive(PKind::Unsigned); }
+  bool isSigned() const { return isPrimitive(PKind::Signed); }
+  bool isFloat() const { return isPrimitive(PKind::Float); }
+
+public:
+  /// The helper for typedef unwrapping.
+  ///
+  /// \returns - nullptr if called with a non-typedef definition, since it's not
+  ///            possible to produce a `model::Type` for one of those without
+  ///            placing it into a binary.
+  ///          - a pointer to an unwrapped `model::Type` in all the other cases.
+  model::Type *skipTypedefs();
+  const model::Type *skipTypedefs() const;
+
+  /// The helper for unwrapping typedefs, including const ones.
+  /// See \ref skipTypedef documentation for specifics.
+  model::Type *skipConstAndTypedefs();
+  const model::Type *skipConstAndTypedefs() const;
```

Comment:
See discussion about `skipTypedefs` vs `resolveTypedefs`.

`skipTypedef`s does not document the fact that it stops at `typedef`s with `IsConst = true`.